src/components/NewTaxonomy/NewTaxonomy.tsx

```diff
@@ -83,14 +83,41 @@ const NewTaxonomy = ({
     setTreeData(convert(items, options));
   }, [items]);
 
+  const flatten = useMemo(() => {
+    const flatten: TaxonomyItem[] = [];
+    const visitItem = (item: TaxonomyItem) => {
+      flatten.push(item);
+      item.children?.forEach(visitItem);
+    };
+
+    items.forEach(visitItem);
+    return flatten;
+  }, [items]);
+
+  const formatOutputValue = useMemo(() => {
+    return selected.map(path => {
+      const selectedItem = path.map(value => {
+        const label = flatten.find(taxonomyItem => taxonomyItem.path.at(-1) === value)?.label;
```

Comment:
By searching for the alias which is not unique through the whole list of items, we can accidentially grab the wrong one. For example with simple ordered numbers as aliases on every level, so the path will be like `["1", "1", "4", "1"]`, we'll get the same item three times, two of them will be the wrong ones.
We should go down the tree as far as we can to find the correct item. This will improve performance (less memory because of dropping `flatten` and fewer passes because of scanning only one branch for every level) and will make it correct. For not loaded items we'll just return the rest of the path.